src/utilities/effects.ts

```diff
@@ -117,29 +128,34 @@ const effects = {
     planId: number,
     numOfUserChanges: number,
   ): Promise<void> {
-    let confirm: boolean = true;
-    if (numOfUserChanges > 0) {
-      ({ confirm } = await showConfirmModal(
-        'Apply Preset',
-        `There ${numOfUserChanges > 1 ? 'are' : 'is'} currently ${numOfUserChanges} manually edited parameter${
-          numOfUserChanges > 1 ? 's' : ''
-        }. This will remove existing edits and apply preset parameters.`,
-        'Apply Preset to Activity Directive',
-      ));
-    }
-
-    if (confirm) {
-      try {
+    try {
+      if (!queryPermissions.APPLY_PRESET_TO_ACTIVITY()) {
```

Comment:
One _slightly_ shorter way to write this could be:
```
!queryPermissions.APPLY_PRESET_TO_ACTIVITY() && throwPermissionError('apply a preset to an activity directive');
```
though I don't know if it's worth the conciseness as it looks sort of odd.

Reply:
ooo...that is kinda slick actually haha. I did struggle with the idea of wrapping these checks in a function since it's a lot of duplicate code, but I ultimately decided that it was more readable this way. Your suggestion is nice, but I do wonder how readable it is. It could just be because I'm not too used to seeing this usage that I don't immediately understand what it's doing, but if you feel strongly enough that this will be commonplace soon, I'd be up for changing it.

Reply:
Nah I agree that it's less readable. An alternative one liner could be some function that takes in a boolean and an error message that would check the boolean and throw the error? Like `checkPermissions(queryPermissions.APPLY_PRESET_TO_ACTIVITY(), 'apply a preset to an activity directive')`?

Reply:
Yeah agreed, let's keep it as is.